src/App.js

```diff
@@ -30,42 +33,98 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState('');
+  const [turns, setTurns] = useState(0)
 
   // Wave 2
   // You will need to create a method to change the square 
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
+  const onClickCallback = (id) => {
+    let row = 0
+    while (id > 2) {
+      id -= 3
+      row++
+    }
 
+    if (!squares[row][id]['value'] && !winner) {
+      const newSquares = [...squares]
+      newSquares[row][id]['value'] = currentPlayer
+      setSquares(newSquares)
 
-  const checkForWinner = () => {
+      const updatedPlayer = currentPlayer === PLAYER_1 ? PLAYER_2 : PLAYER_1
+      setCurrentPlayer(updatedPlayer)
+      const gameOver = checkForWinner(row, id)
+      if (!gameOver) {
+        if (turns === 8) {
+          setWinner('Tie')
+        } else {
+          setTurns(turns + 1)
+        }
+      }
+    }
+  }
+
+  const checkForWinner = (row, col) => {
     // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if 
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if 
-    //    all three squares have the same value.
 
+    // check row
+    if (squares[row][0]['value'] === currentPlayer &&
```

Comment:
Nice optimizations in checking for the winner. You could go a step further in condensing the code by looping over the number of columns here (the number of rows later).